### DISCUSSION TITLE
Is it possible to make playwright python version?

### DISCUSSION THREAD

**Discussion by Guria** [RecordReplay/devtools#8768] [Q&A]
Is there a way to integrate replay.io with python version of playwright?
As I can understand it just needs custom version of playwright headless version binaries.
Would appreciate if you can suggest potential directions to hack into.

**Comment by jasonLaster** [comment]
Hey @Guria this should be relatively easy because our Playwright CLI just installs the binary
https://github.com/replayio/replay-cli/tree/main/packages/playwright

CC @ryanjduffy knows how this stuff works and can guide. Perhaps a good next step would be to start a replay playwright python repo that we can hack on?

Also, what's your use case? And please join our replay.io/discord where we can discuss

**Comment by ryanjduffy** [comment]
It should be possible. The main integration point is [setting the executable path](https://playwright.dev/python/docs/api/class-browsertype#browser-type-launch) to the replay browser. We expose a method for this in the node API but it's pretty easy to construct manually.

You can install the browsers with our NPM module:

```sh
# Install the replay browsers ... typically in $HOME/.replay/playwright
npx @replayio/playwright install
```

The specific paths to the browser executable will vary based on the browser and platform but take a look at the [install script](https://github.com/replayio/replay-cli/blob/471d967b69e3ffba3f8a6d3165dabe4af1c02aee/packages/replay/src/install.ts) to see how they're constructed.
